### PR TITLE
Fix some documentation about the category channels

### PR DESCRIPTION
**Pull request by KingDGrizzle**
**Please describe the changes this PR makes and why it should be merged:**
Just some doc fixes

Gonna check everything now..

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [x] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/Guild.js [hunk 1]
```diff
@@ -895,7 +895,7 @@ class Guild extends Base {
    * @param {Object} [options] Options
    * @param {Array<PermissionOverwrites|ChannelCreationOverwrites>} [options.overwrites] Permission overwrites
    * @param {string} [options.reason] Reason for creating this channel
-   * @returns {Promise<TextChannel|VoiceChannel>}
+   * @returns {Promise<CategoryChannel|TextChannel|VoiceChannel>}
```
Comment (devsnek) [inline src/structures/Guild.js]: Just make it GuildChannel

File: src/structures/GuildChannel.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ const Constants = require('../util/Constants');
 const { TypeError } = require('../errors');
 
 /**
- * Represents a guild channel (i.e. text channels and voice channels).
+ * Represents a guild channel (i.e. text channels, category channels and voice channels).
```
Comment (devsnek) [inline src/structures/GuildChannel.js]: This is an exemplitive statement so it doesn't need to include all cases

File: src/structures/GuildChannel.js [hunk 3]
```diff
@@ -59,6 +59,7 @@ class GuildChannel extends Channel {
   /**
    * The category parent of this channel
    * @type {?GuildChannel}
```
Comment (devsnek) [inline src/structures/GuildChannel.js]: Change to ?CategoryChannel

File: src/structures/GuildChannel.js [hunk 4]
```diff
@@ -309,7 +312,7 @@ class GuildChannel extends Channel {
   /**
    * Set the category parent of this channel.
    * @param {GuildChannel|Snowflake} channel Parent channel
-   * @param {boolean} [options.lockPermissions] Lock the permissions to what the parent's permissions are
+   * @param {boolean} [options.lockPermissions] Lock the permissions to what the parents permissions are
```
Comment (devsnek) [inline src/structures/GuildChannel.js]: This is wrong

Comment (KingDGrizzle) [reply src/structures/GuildChannel.js]: Its wrong due to the `parents` or? Should I just revert it?

Comment (robflop) [reply src/structures/GuildChannel.js]: It's grammatically wrong. You should revert it.

File: src/structures/GuildChannel.js (hunk 2 above)
Comment (devsnek) [reply src/structures/GuildChannel.js]: And change it to e.g.

**Pull request merged**